cmake error for v3.6.4 with CPU backend

Hey!

I am trying to build `v3.6.4` with CPU backend in the Dockerfile 
```
FROM ubuntu:16.04

RUN APT_INSTALL="apt-get install -y --no-install-recommends" && \
    apt-get update && \
    DEBIAN_FRONTEND=noninteractive $APT_INSTALL \
        build-essential \
        ca-certificates \
        wget \
        git \
        vim \
        emacs \
        nano \
        htop \
        g++ \
        # for MKL
        apt-transport-https \
        # for arrayfire CPU backend
        # OpenBLAS
        libopenblas-dev libfftw3-dev liblapacke-dev \
        # ATLAS
        libatlas3-base libatlas-dev libfftw3-dev liblapacke-dev \
        # ssh for OpenMPI
        openssh-server openssh-client \
        # OpenMPI
        libopenmpi-dev libomp-dev && \
# ==================================================================
# cmake 3.10 (for MKLDNN)
# ------------------------------------------------------------------
    apt-get purge -y cmake && \
    # for cmake
    DEBIAN_FRONTEND=noninteractive $APT_INSTALL zlib1g-dev libcurl4-openssl-dev && \
    cd /tmp && wget https://cmake.org/files/v3.10/cmake-3.10.3.tar.gz  && \
    tar -xzvf cmake-3.10.3.tar.gz  && cd cmake-3.10.3  && \
    ./bootstrap --system-curl && \
    make -j8 &&  make install && cmake --version && \
# ==================================================================
# arrayfire with CPU backend https://github.com/arrayfire/arrayfire/wiki/
# ------------------------------------------------------------------
    cd /tmp && git clone --recursive https://github.com/arrayfire/arrayfire.git && \
    cd arrayfire && git checkout v3.6.4  && git submodule update --recursive --remote && \
    mkdir build && cd build && \
    CXXFLAGS=-DOS_LNX cmake .. -DCMAKE_BUILD_TYPE=Release -DAF_BUILD_CUDA=OFF -DAF_BUILD_OPENCL=OFF -DAF_BUILD_EXAMPLES=OFF && \
    make -j8 && make install && \
```

with cmake configuration:
```
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for CL_VERSION_2_0
-- Looking for CL_VERSION_2_0 - not found
-- Looking for CL_VERSION_1_2
-- Looking for CL_VERSION_1_2 - not found
-- Looking for CL_VERSION_1_1
-- Looking for CL_VERSION_1_1 - not found
-- Looking for CL_VERSION_1_0
-- Looking for CL_VERSION_1_0 - not found
-- Could NOT find OpenCL (missing: OpenCL_LIBRARY OpenCL_INCLUDE_DIR)
CUDA_TOOLKIT_ROOT_DIR not found or specified
-- Could NOT find CUDA (missing: CUDA_TOOLKIT_ROOT_DIR CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY) (Required is at least version "7.0")                                                     -- Could NOT find OpenCL (missing: OpenCL_LIBRARY OpenCL_INCLUDE_DIR) (Required is at least version "1.2")                                                                                                  -- Could NOT find OpenGL (missing: OPENGL_gl_LIBRARY OPENGL_INCLUDE_DIR)                                                                                                                                    -- Found OpenMP_C: -fopenmp (found version "4.0")                                                                                                                                                           -- Found OpenMP_CXX: -fopenmp (found version "4.0")
-- Could NOT find FreeImage (missing: FreeImage_INCLUDE_DIR FreeImage_LINK_LIBRARY)                                                                                                                         -- Looking for pthread.h                                                                                                                                                                                    -- Looking for pthread.h - found                                                                                                                                                                            -- Looking for pthread_create                                                                                                                                                                               -- Looking for pthread_create - not found                                                                                                                                                                   -- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
-- Found FFTW: /usr/include
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of void*
-- Check size of void* - done
-- Checking for [Accelerate]
-- Checking for [vecLib]
-- Checking for [cblas - atlas]
-- Includes found
-- Looking for cblas_dgemm
-- Looking for cblas_dgemm - found
-- CBLAS Symbols FOUND
-- CBLAS library found
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
-- Found LAPACK: /usr/include
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Check size of int
-- Check size of int - done
-- Found MKL: /opt/intel/mkl/include
-- Could NOT find Boost
-- BOOST_COMPUTE: /tmp/arrayfire/build/boost_compute-prefix/src/boost_compute
-- Performing Test has_ignored_attributes_flag
-- Performing Test has_ignored_attributes_flag - Success
-- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE)
-- Configuring done
```
and sometimes it crashes with the following error
```
/tmp/arrayfire/src/backend/common/Logger.cpp:29:15: error: 'spdlog::get' has not been declared
 using spdlog::get;
               ^
/tmp/arrayfire/src/backend/common/Logger.cpp:31:15: error: 'spdlog::stdout_logger_mt' has not been declared
 using spdlog::stdout_logger_mt;
               ^
/tmp/arrayfire/src/backend/common/Logger.cpp: In function 'std::shared_ptr<spdlog::logger> common::loggerFactory(std::__cxx11::string)':
/tmp/arrayfire/src/backend/common/Logger.cpp:37:28: error: no matching function for call to 'get(std::__cxx11::string&)'
     if (!(logger = get(name))) {
                            ^
In file included from /usr/include/c++/5/functional:55:0,
                 from /usr/include/c++/5/memory:79,
                 from /tmp/arrayfire/src/backend/common/Logger.hpp:12,
                 from /tmp/arrayfire/src/backend/common/Logger.cpp:14:
/usr/include/c++/5/tuple:844:5: note: candidate: template<long unsigned int __i, class ... _Elements> constexpr std::__tuple_element_t<__i, std::tuple<_Elements ...> >&& std::get(std::tuple<_Elements ...>
&&)
     get(tuple<_Elements...>&& __t) noexcept
     ^
/usr/include/c++/5/tuple:844:5: note:   template argument deduction/substitution failed:
/tmp/arrayfire/src/backend/common/Logger.cpp:37:28: note:   'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' is not derived from 'std::tuple<_Elements ...>'
     if (!(logger = get(name))) {
                            ^
In file included from /usr/include/c++/5/functional:55:0,
                 from /usr/include/c++/5/memory:79,
                 from /tmp/arrayfire/src/backend/common/Logger.hpp:12,
                 from /tmp/arrayfire/src/backend/common/Logger.cpp:14:
/usr/include/c++/5/tuple:838:5: note: candidate: template<long unsigned int __i, class ... _Elements> constexpr std::__tuple_element_t<__i, std::tuple<_Elements ...> >& std::get(const std::tuple<_Elements
 ...>&)
     get(const tuple<_Elements...>& __t) noexcept
     ^
/usr/include/c++/5/tuple:838:5: note:   template argument deduction/substitution failed:
/tmp/arrayfire/src/backend/common/Logger.cpp:37:28: note:   'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' is not derived from 'const std::tuple<_Elements ...>'
     if (!(logger = get(name))) {
In file included from /usr/include/c++/5/functional:55:0,
                 from /usr/include/c++/5/memory:79,
                 from /tmp/arrayfire/src/backend/common/Logger.hpp:12,
                 from /tmp/arrayfire/src/backend/common/Logger.cpp:14:
/usr/include/c++/5/tuple:832:5: note: candidate: template<long unsigned int __i, class ... _Elements> constexpr std::__tuple_element_t<__i, std::tuple<_Elements ...> >& std::get(std::tuple<_Elements ...>&
)
     get(tuple<_Elements...>& __t) noexcept
     ^
/usr/include/c++/5/tuple:832:5: note:   template argument deduction/substitution failed:
/tmp/arrayfire/src/backend/common/Logger.cpp:37:28: note:   'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' is not derived from 'std::tuple<_Elements ...>'
     if (!(logger = get(name))) {
                            ^
In file included from /usr/include/c++/5/tuple:39:0,
                 from /usr/include/c++/5/functional:55,
                 from /usr/include/c++/5/memory:79,
                 from /tmp/arrayfire/src/backend/common/Logger.hpp:12,
                 from /tmp/arrayfire/src/backend/common/Logger.cpp:14:
/usr/include/c++/5/array:297:5: note: candidate: template<long unsigned int _Int, class _Tp, long unsigned int _Nm> constexpr const _Tp& std::get(const std::array<_Tp, _Nm>&)
     get(const array<_Tp, _Nm>& __arr) noexcept
     ^
/usr/include/c++/5/array:297:5: note:   template argument deduction/substitution failed:
/tmp/arrayfire/src/backend/common/Logger.cpp:37:28: note:   'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' is not derived from 'const std::array<_Tp, _Nm>'
     if (!(logger = get(name))) {
                            ^
In file included from /usr/include/c++/5/tuple:39:0,
                 from /usr/include/c++/5/functional:55,
                 from /usr/include/c++/5/memory:79,
                 from /tmp/arrayfire/src/backend/common/Logger.hpp:12,
                 from /tmp/arrayfire/src/backend/common/Logger.cpp:14:
/usr/include/c++/5/array:289:5: note: candidate: template<long unsigned int _Int, class _Tp, long unsigned int _Nm> constexpr _Tp&& std::get(std::array<_Tp, _Nm>&&)
     get(array<_Tp, _Nm>&& __arr) noexcept
     ^
/usr/include/c++/5/array:289:5: note:   template argument deduction/substitution failed:
/tmp/arrayfire/src/backend/common/Logger.cpp:37:28: note:   'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' is not derived from 'std::array<_Tp, _Nm>'
     if (!(logger = get(name))) {
....
/tmp/arrayfire/src/backend/common/Logger.cpp:38:39: error: 'stdout_logger_mt' was not declared in this scope
         logger = stdout_logger_mt(name);
                                       ^
/tmp/arrayfire/src/backend/common/Logger.cpp:39:15: error: invalid use of incomplete type 'class spdlog::logger'
         logger->set_pattern("[%n][%t] %v");
               ^
In file included from /tmp/arrayfire/src/backend/common/Logger.cpp:14:0:
/tmp/arrayfire/src/backend/common/Logger.hpp:16:7: note: forward declaration of 'class spdlog::logger'
 class logger;
       ^
/tmp/arrayfire/src/backend/common/Logger.cpp:45:19: error: invalid use of incomplete type 'class spdlog::logger'
             logger->set_level(trace);
                   ^
In file included from /tmp/arrayfire/src/backend/common/Logger.cpp:14:0:
/tmp/arrayfire/src/backend/common/Logger.hpp:16:7: note: forward declaration of 'class spdlog::logger'
 class logger;
       ^
[  6%] Building CXX object src/api/unified/CMakeFiles/af.dir/__/__/backend/common/util.cpp.o
At global scope:
cc1plus: warning: unrecognized command line option '-Wno-ignored-attributes'
src/api/unified/CMakeFiles/af.dir/build.make:614: recipe for target 'src/api/unified/CMakeFiles/af.dir/__/__/backend/common/Logger.cpp.o' failed
make[2]: *** [src/api/unified/CMakeFiles/af.dir/__/__/backend/common/Logger.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  6%] Building CXX object src/backend/cpu/CMakeFiles/afcpu.dir/approx.cpp.o
[  6%] Building CXX object src/backend/cpu/CMakeFiles/afcpu.dir/assign.cpp.o
[  6%] Building CXX object src/backend/cpu/CMakeFiles/afcpu.dir/bilateral.cpp.o
[  6%] Building CXX object src/backend/cpu/CMakeFiles/afcpu.dir/blas.cpp.o
[  6%] Building CXX object src/backend/cpu/CMakeFiles/afcpu.dir/canny.cpp.o
[  6%] Building CXX object src/backend/cpu/CMakeFiles/afcpu.dir/cholesky.cpp.o
make[1]: *** [src/api/unified/CMakeFiles/af.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Any idea how to fix this?

Thanks for help!

--- a/src/backend/common/Logger.hpp
+++ b/src/backend/common/Logger.hpp
@@ -11,10 +11,8 @@
 
 #include <memory>
 #include <string>
+#include <spdlog/spdlog.h>
 
-namespace spdlog {
-class logger;
-}
 namespace common {
 std::shared_ptr<spdlog::logger> loggerFactory(std::string name);
 std::string bytesToString(size_t bytes);

